lib/moon/helpers/surface_render.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Moon.Helpers.SurfaceRender do
+  @moduledoc """
+  Functions for rendering Surface components in HEEX/SLIM/... (not Surface) templates
+  """
+
+  import Phoenix.LiveView.Helpers, only: [component: 2, live_component: 1]
+
+  defp get_default_props(module) do
+    Enum.reduce(module.__props__(), %{__context__: %{}}, fn
+      %{name: name, opts: opts}, acc ->
+        Map.put(acc, name, Keyword.get(opts, :default))
+    end)
+  end
+
+  defp transform_slots(
+         props = %{inner_block: [%{__slot__: :inner_block, inner_block: inner_block}]}
+       ) do
+    props
+    |> Map.delete(:inner_block)
+    |> Map.put(:default, [%{__slot__: :default, inner_block: inner_block}])
+  end
+
+  defp transform_slots(props), do: props
+
+  @doc "used for rendering 'dead'/stateless Surface componet"
+  def surface_component(module, props) do
+    component(
+      &module.render/1,
+      module |> get_default_props() |> Map.merge(props |> transform_slots())
+    )
+  end
+
+  def surface_component(props = %{module: module}),
+    do: surface_component(module, props |> Map.delete(:module))
+
+  @doc "used for rendering live/stateful Surface componet"
+  def surface_live_component(props = %{module: module}) do
+    module |> get_default_props() |> Map.merge(props |> transform_slots()) |> live_component()
+  end
+
+  def surface_live_component(module, props),
+    do: surface_live_component(Map.put(props, :module, module))
+
+  defp get_render_function(module) do
+    %{kind: :component} = module.__live__()
+    &surface_live_component/1
+  rescue
+    [UndefinedFunctionError, MatchError] ->
+      &surface_component/1
+  end
+
+  @doc """
+  A function for rendering Surface components inside non-surface templates
+  usage: `<.surface module="Button">hit me!</.surface>`
+  Please note that only limited functional is supported, e.g. contexts and named
+  slots are out of scope, sorry.
+  """
+  def surface(props = %{module: module}) do
+    get_render_function(module).(props)
+  end
+
+  # In older versions of phoenex_live_view (e.g. 0.17.6 but prior to 0.17.12) there is a bug
```

Comment:
This bug is fixed? If not, let's move to latest LiveView and Surfae.

Reply:
this code is no longer needed when using phoenix_live_view v.0.17.12 (our current), but bug still exist in 0.17.6, so may be useful for one who used it.

Reply:
otherwise i can set phoenix_live_view ~> 0.17.12 in mix.exs - than code can be deleted